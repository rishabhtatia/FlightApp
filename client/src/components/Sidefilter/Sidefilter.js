import React, { useState } from 'react';
import styles from './Sidefilter.module.css';
import _ from 'lodash';
import PropTypes from 'prop-types';
import clsx from 'clsx';
import moment from 'moment';

const SideFilter = (props) => {
  const [formData, setFormData] = useState({
    originCity: '',
    destinationCity: '',
    departureDate: '',
    returnDate: '',
  });

  const handleInput = (event) => {
    setFormData((prevState) => {
      return { ...prevState, [event.target.name]: event.target.value };
    });
  };

  const isDisabled = () => {
    return (
      !moment(formData.departureDate, 'YYYY-MM-DD').isValid() ||
      !moment(formData.returnDate, 'YYYY-MM-DD').isValid() ||
      !formData?.originCity?.length > 0 ||
      !formData?.destinationCity?.length > 0 ||
      formData.originCity === formData.destinationCity
    );
  };

  return (
    <div className={styles.sideFilter}>
      <div className={styles.buttonGroup}>
        <button
          className={clsx('btn', styles.button, {
            'btn-primary': !props.showReturnData,
          })}
          onClick={() => props.journeyType(false)}
        >
          One Way
        </button>
        <button
          className={clsx('btn', styles.button, {
            'btn-primary': props.showReturnData,
          })}
          onClick={() => props.journeyType(true)}
        >
          Return
        </button>
      </div>
      <form onSubmit={(event) => props.onSearch(event, formData)}>
        <div className="form-group">
          <select
            name="originCity"
            className={'mdb-select md-form' + styles.inputGroup}
            searchable="Search here.."
            value={formData.originCity}
            onChange={handleInput}
          >
            <option value="" disabled selected>
              Enter Origin City
            </option>
            {_.map(props.dropdownList, (item) => {
              return (
                <option value={item.value} key={item.value}>
                  {item.value}
                </option>
              );
            })}
          </select>
        </div>
        <div className="form-group">
          <select
            name="destinationCity"
            className={'mdb-select md-form' + styles.inputGroup}
            searchable="Search here.."
            value={formData.destinationCity}
            onChange={handleInput}
          >
            <option value="" disabled selected>
              Enter Destination City
            </option>
            {_.map(props.dropdownList, (item) => {
              return (
                <option value={item.value} key={item.value}>
                  {item.value}
                </option>
              );
            })}
          </select>
        </div>
        <div className="form-group">
          <input
            type="date"
            className={styles.inputGroup}
            name="departureDate"
            value={formData.departureDate}
            onChange={handleInput}
          ></input>
        </div>
        <div className="form-group">
          <input
            type="date"
            className={styles.inputGroup}
            name="returnDate"
            value={formData.returnDate}
            onChange={handleInput}
          ></input>
        </div>
        <button
          type="submit"
          className="btn btn-primary"
          disabled={isDisabled()}
        >
          Search
        </button>
      </form>
    </div>
  );
};

SideFilter.defaultpropTypes = {
  dropdownList: [],
  showReturnData: false,
};

SideFilter.propTypes = {
  dropdownList: PropTypes.array,
  onSearch: PropTypes.func.isRequired,
  journeyType: PropTypes.func.isRequired,
  showReturnData: PropTypes.bool,
};

export default SideFilter;
