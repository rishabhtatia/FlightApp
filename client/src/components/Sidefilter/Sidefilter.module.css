.buttonGroup {
  align-items: right;
  margin: 5px;
}
.button {
  margin: 5px;
  padding: 5px 10px;
  cursor: pointer;
}
.sideFilter {
  border: 1px solid black;
  margin: 0 5px;
  width: 100%;
  height: 100%;
}
.inputGroup {
  box-sizing: border-box;
  width: 80%;
}
