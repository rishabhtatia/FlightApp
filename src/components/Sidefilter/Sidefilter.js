import React, { useState } from 'react';
import styles from './Sidefilter.module.css';
import _ from 'lodash';
import clsx from 'clsx';
import moment from 'moment';

const SIDEFILTER = (props) => {
  const [formData, formDataHandler] = useState({
    originCity: '',
    destinationCity: '',
    departureDate: '',
    returnDate: '',
  });

  const handleInput = (event) => {
    formDataHandler((prevState) => {
      return { ...prevState, [event.target.name]: event.target.value };
    });
  };

  const isDisabled = () => {
    return (
      !moment(formData.departureDate, 'YYYY-MM-DD').isValid() ||
      !moment(formData.returnDate, 'YYYY-MM-DD').isValid() ||
      !formData.originCity.length > 0 ||
      !formData.destinationCity.length > 0 ||
      formData.originCity === formData.destinationCity
    );
  };

  return (
    <div className={styles.sideFilter}>
      <div className={styles.buttonGroup}>
        <button
          className={clsx('btn', styles.button, {
            'btn-primary': !props.showReturnData,
          })}
          onClick={() => props.journeyType(false, formData)}
        >
          One Way
        </button>
        <button
          className={clsx('btn', styles.button, {
            'btn-primary': props.showReturnData,
          })}
          onClick={() => props.journeyType(true, formData)}
        >
          Return
        </button>
      </div>
      <form onSubmit={(event) => props.onSearch(event, formData)}>
        <div className="form-group">
          <select
            name="originCity"
            className={'mdb-select md-form' + styles.inputGroup}
            searchable="Search here.."
            value={formData.originCity}
            onChange={handleInput}
          >
            <option value="" disabled selected>
              Enter Origin City
            </option>
            {_.map(props.dropdownList, (item) => {
              return (
                <option value={item.value} key={item.value}>
                  {item.value}
                </option>
              );
            })}
          </select>
        </div>
        <div className="form-group">
          <select
            name="destinationCity"
            className={'mdb-select md-form' + styles.inputGroup}
            searchable="Search here.."
            value={formData.destinationCity}
            onChange={handleInput}
          >
            <option value="" disabled selected>
              Enter Destination City
            </option>
            {_.map(props.dropdownList, (item) => {
              return (
                <option value={item.value} key={item.value}>
                  {item.value}
                </option>
              );
            })}
          </select>
        </div>
        <div className="form-group">
          <input
            type="date"
            className={styles.inputGroup}
            name="departureDate"
            value={formData.departureDate}
            onChange={handleInput}
          ></input>
        </div>
        <div className="form-group">
          <input
            type="date"
            className={styles.inputGroup}
            name="returnDate"
            value={formData.returnDate}
            onChange={handleInput}
          ></input>
        </div>
        <button
          type="submit"
          className="btn btn-primary"
          disabled={isDisabled()}
        >
          Search
        </button>
      </form>
    </div>
  );
};

export default SIDEFILTER;
