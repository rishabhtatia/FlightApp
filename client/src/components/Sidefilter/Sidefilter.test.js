import React from 'react';
import renderer from 'react-test-renderer';

import Sidefilter from './Sidefilter';

it('rendering with nothing passed', () => {
  const tree = renderer.create(<Sidefilter />).toJSON();
  expect(tree).toMatchSnapshot();
});
