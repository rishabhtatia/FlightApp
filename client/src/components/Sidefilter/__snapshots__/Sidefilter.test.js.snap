// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with nothing passed 1`] = `
<div
  className="sideFilter"
>
  <div
    className="buttonGroup"
  >
    <button
      className="btn button btn-primary"
      onClick={[Function]}
    >
      One Way
    </button>
    <button
      className="btn button"
      onClick={[Function]}
    >
      Return
    </button>
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <select
        className="mdb-select md-forminputGroup"
        name="originCity"
        onChange={[Function]}
        searchable="Search here.."
        value=""
      >
        <option
          disabled={true}
          selected={true}
          value=""
        >
          Enter Origin City
        </option>
      </select>
    </div>
    <div
      className="form-group"
    >
      <select
        className="mdb-select md-forminputGroup"
        name="destinationCity"
        onChange={[Function]}
        searchable="Search here.."
        value=""
      >
        <option
          disabled={true}
          selected={true}
          value=""
        >
          Enter Destination City
        </option>
      </select>
    </div>
    <div
      className="form-group"
    >
      <input
        className="inputGroup"
        name="departureDate"
        onChange={[Function]}
        type="date"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="inputGroup"
        name="returnDate"
        onChange={[Function]}
        type="date"
        value=""
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;